### PR TITLE
Implement String.prototype.indexOf()

### PR DESCRIPTION
**Pull request by lvidacs**
JerryScript-DCO-1.0-Signed-off-by: Laszlo Vidacs lvidacs.u-szeged@partner.samsung.com

### CODE REVIEW

File: jerry-core/ecma/builtin-objects/ecma-builtin-string-prototype.cpp [hunk 1]
```diff
@@ -307,7 +307,122 @@ ecma_builtin_string_prototype_object_index_of (ecma_value_t this_arg, /**< this
...
+
+  /* 4b, 6 */
+  ecma_length_t start = ecma_builtin_helper_string_index_normalize (pos_num, original_len);
+
+  /* 7 */
+  ecma_string_t *search_str_p = ecma_get_string_from_value (search_str_val);
+  const ecma_length_t search_len = ecma_string_get_length (search_str_p);
+  const lit_utf8_size_t search_size = ecma_string_get_size (search_str_p);
+
+  ecma_number_t * ret_num = ecma_alloc_number ();
```
Comment (galpeter) [inline jerry-core/ecma/builtin-objects/ecma-builtin-string-prototype.cpp]: No need for space after the `*`

**Comment by lvidacs** [comment]
@galpeter, thanks for the review, the code is updated also with the _p suffix.

**Comment by egavrin** [comment]
Good to me

**Comment by galpeter** [comment]
lgtm

**Comment by galpeter** [comment]
Landed: https://github.com/Samsung/jerryscript/commit/64c81cc44d288427e4e04349afab1a99b23e3e54

**Pull request closed without merging**